README.md

```diff
@@ -1 +1,284 @@
-# nnf-container-example
\ No newline at end of file
+# NNF Container Workflow Tutorial
+
+## Overview
+
+This repo contains an example MPI application that is built into a container that can be used for an
+NNF Container Workflow. The application uses the dynamic storage created by the workflow. The
+storage path is passed into the hello world application.
+
+The overall process for creating a container workflow is as follows:
+
+* A container image is created from this repo
+* An NNF Container Profile is created on the NNF Kubernetes cluster:
+  * The container image is specified in the profile along with the command to run
+  * A required GFS2 storage is defined in the profile
+  * The command also includes an environment
+* A Workflow is created and contains two directives:
+  * Directive for the GFS2 filesystem (that matches the GFS2 storage name in the profile)
+  * Directive for the container (that specifies the profile)
+* Workflow is progressed to the PreRun stage, where the container(s) starts
+
+## Making a Container Image
+
+To start, you must have a working container image that includes your application. This image and
+your application are used in the NNF Container Profile to instruct the container workflow to run
+your application.
+
+The `Dockerfile` in this repository creates an example image that can be used to drive container
+workflows.
+
+When building your own image, ensure it meets the following requirements coupled with your user
+application.
+
+### Requirements
+
+Any container image that is built must be available in an image registry that is available on your
+cluster. See your cluster administrator for more details.
+
+In this example, we're using the GitHut Container Registry (ghcr.io), so your cluster must have internet access to retrieve the image.
+
+#### MPI Applications
+
+For MPI applications, the container must include the following:
+
+* open-mpi
+* MPI File Utils
+* ssh server
+
+The easiest way to do this is to use the NNF MFU (MPI File Utils) image in your Dockerfile:
+
+```dockerfile
+FROM ghcr.io/nearnodeflash/nnf-mfu:latest
```

Comment:
You could switch that "latest" to "master" or "0.0.1"...    The "master" tag tracks the most recent build from the master branch.  The "latest" tag tracks the most recent build from any releases/v* branch.
Maybe "0.0.1" would be the better choice.

Reply:
I strugged with this. At first, I thought I'd want `master`, so that we don't have to update these docs when things change. But at the same time, I don't want the rug pulled out underneath.

I don't think nnf-mfu will change much, so might be splitting hairs here?

Reply:
My objective is to make sure you're aware of what's happening with the image tags now that I've added the release workflow, and being mindful about which one you're choosing.

I think most people might expect the latest build from the master branch.  They can fuss with specific releases on their own.

Reply:
(That's me, waffling...)

Reply:
Let's go with master then!